ClangTidy warnings for `misc-const-correctness`

```
/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:570:3: error: variable 'frame' of type 'sourcemeta::core::SchemaFrame' can be declared 'const' [misc-const-correctness,-warnings-as-errors]
  570 |   sourcemeta::core::SchemaFrame frame{
      |   ^
      |                                 const

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:650:3: error: variable 'frame' of type 'sourcemeta::core::SchemaFrame' can be declared 'const' [misc-const-correctness,-warnings-as-errors]
  650 |   sourcemeta::core::SchemaFrame frame{
      |   ^
      |                                 const

/Users/jviotti/Projects/core/src/core/jsonschema/resolver.cc:23:3: error: variable 'frame' of type 'SchemaFrame' can be declared 'const' [misc-const-correctness,-warnings-as-errors]
   23 |   SchemaFrame frame{SchemaFrame::Mode::References};
      |   ^
      |               const

/Users/jviotti/Projects/core/src/core/jsonschema/resolver.cc:27:3: error: variable 'added_any_schema' of type 'bool' can be declared 'const' [misc-const-correctness,-warnings-as-errors]
   27 |   bool added_any_schema{false};
      |   ^
      |        const

/Users/jviotti/Projects/core/src/core/jsonschema/transformer.cc:103:3: error: variable 'frame' of type 'SchemaFrame' can be declared 'const' [misc-const-correctness,-warnings-as-errors]
  103 |   SchemaFrame frame{SchemaFrame::Mode::Locations};
      |   ^
      |               const

/Users/jviotti/Projects/core/src/core/jsonschema/transformer.cc:106:3: error: variable 'result' of type 'bool' can be declared 'const' [misc-const-correctness,-warnings-as-errors]
  106 |   bool result{true};
      |   ^
      |        const

/Users/jviotti/Projects/core/src/core/jsonschema/transformer.cc:149:3: error: variable 'result' of type 'bool' can be declared 'const' [misc-const-correctness,-warnings-as-errors]
  149 |   bool result{true};
      |   ^
      |        const

/Users/jviotti/Projects/core/src/core/jsonschema/transformer.cc:151:5: error: variable 'frame' of type 'SchemaFrame' can be declared 'const' [misc-const-correctness,-warnings-as-errors]
  151 |     SchemaFrame frame{SchemaFrame::Mode::References};
      |     ^
      |                 const

/Users/jviotti/Projects/core/src/core/jsonschema/transformer.cc:154:5: error: variable 'applied' of type 'bool' can be declared 'const' [misc-const-correctness,-warnings-as-errors]
  154 |     bool applied{false};
      |     ^
      |          const

/Users/jviotti/Projects/core/src/extension/alterschema/simplify/dependencies_property_tautology.h:45:7: error: variable 'match' of type 'bool' can be declared 'const' [misc-const-correctness,-warnings-as-errors]
   45 |       bool match{false};
      |       ^
      |            const

/Users/jviotti/Projects/core/src/extension/alterschema/simplify/dependent_required_tautology.h:38:7: error: variable 'match' of type 'bool' can be declared 'const' [misc-const-correctness,-warnings-as-errors]
   38 |       bool match{false};
      |       ^
      |            const

```

--- a/cmake/common/targets/clang-tidy.config
+++ b/cmake/common/targets/clang-tidy.config
@@ -11,7 +11,7 @@ Checks: '-*,
           performance-*,
           portability-*,
           objc-*,
-          misc-*,-misc-no-recursion'
+          misc-*,-misc-no-recursion,-misc-const-correctness'
 WarningsAsErrors: '*'
 HeaderFilterRegex: ''
 FormatStyle: none

